fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -280,13 +277,9 @@ public Response deleteObject() {
 
         try {
             resource().delete();
-
-            try {
-                session.save();
-            } catch (final RepositoryException e) {
-                throw new RepositoryRuntimeException(e);
+            if (!batchService.exists(session.getId(), getUserPrincipal())) {
```

Comment:
Thanks for highlighting the distributed scenario. As for the reflection, that weirdness held the beauty of encapsulating the logic of whether to `Session.save` or not.